ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java

```diff
@@ -77,25 +75,26 @@ private static void applyDeltas(final MutableBeaconState state, final Deltas att
    */
   @Override
   public BeaconState processEpoch(final BeaconState preState) throws EpochProcessingException {
+    return preState.updated(mutableState -> processEpoch(preState, mutableState));
+  }
+
+  protected void processEpoch(final BeaconState preState, final MutableBeaconState state)
+      throws EpochProcessingException {
     final ValidatorStatuses validatorStatuses =
         validatorStatusFactory.createValidatorStatuses(preState);
-    return preState.updated(
-        state -> {
-          processJustificationAndFinalization(state, validatorStatuses.getTotalBalances());
-          processRewardsAndPenalties(state, validatorStatuses);
-          processRegistryUpdates(state, validatorStatuses.getStatuses());
-          processSlashings(state, validatorStatuses.getTotalBalances().getCurrentEpoch());
-          processFinalUpdates(state);
-        });
+    processJustificationAndFinalization(state, validatorStatuses.getTotalBalances());
+    processRewardsAndPenalties(state, validatorStatuses);
+    processRegistryUpdates(state, validatorStatuses.getStatuses());
+    processSlashings(state, validatorStatuses.getTotalBalances().getCurrentEpoch());
+    processEth1DataReset(state);
+    processEffectiveBalanceUpdates(state);
+    processSlashingsReset(state);
+    processRandaoMixesReset(state);
+    processHistoricalRootsUpdate(state);
+    processParticipationUpdates(state);
```

Comment:
v1.1 of the spec split up `processFinalUpdates` for both phase 0 and Altair so mirror that here.